src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import Dropdown from '../dropdown/Dropdown';
+import range from 'lodash/utility/range';
+import flatten from 'lodash/array/flatten';
+
+/*
+  Options is defined here cause we need to not change the ref.
+  This happend because of a very wired behaviour of react-select v0.6.12
+*/
+const options = [];
+
+const H24 = '24h';
+const H12 = '12h';
+const symbolRegex = /[\s,\.\|\/\\]/;
+const numberRegex = /^\d+$/;
+const time12Regex = /^([1-9]|1[0-2])[:,\.\|\/\\]?([0-5]\d)?\s*?(AM|PM)$/;
+const time24Regex = /^([01]?\d|2[0-3])[:,\.\|\/\\]?([0-5]?\d)?$/;
+const separator = ':';
+
+const pad = (num) => num <= 9 ? `0${num}` : num;
+const timeFormatter24 = (hour, minute) => `${pad(hour)}:${pad(minute)}`;
+const timeFormatter12 = (hour, minute) => {
```

Comment:
This function is the very bad. Especially the first if statement, this is because there is more than one logic that have to be checked because it can have as input different things, it looks very specific but actually it work very generally. The first line manage the 0 case (in the optionList generation) and the 12+12 case for the custom generated list.
The second if can be removed even if i make a lot of tries and i didn't find out a way of remove it. The other 2 conditions are pretty obvious.